src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -127,32 +127,29 @@ public static string GetExtension(string path)
         [Pure]
         public static string GetFileName(string path)
         {
-            if (path != null)
-            {
-                PathInternal.CheckInvalidPathChars(path);
-
-                int length = path.Length;
-                for (int i = length - 1; i >= 0; i--)
-                {
-                    char ch = path[i];
-                    if (IsDirectoryOrVolumeSeparator(ch))
-                        return path.Substring(i + 1, length - i - 1);
-                }
-            }
-            return path;
+            if (path == null)
+                return null;
+            PathInternal.CheckInvalidPathChars(path);
```

Comment:
If you move this call into FindFileNameIndex, then it doesn't need to be duplicated across the two call sites.